### PR TITLE
[Gluten-1.2] Backport facebookincubator#9025 and facebookincubator#10979 to fix the result mismatch in RowsStreamingWindowBuild

### PR DESCRIPTION
**Pull request by ccat3z**
Backport https://github.com/facebookincubator/velox/pull/9025 and https://github.com/facebookincubator/velox/pull/10979 to fix https://github.com/apache/incubator-gluten/issues/7194.

1. https://github.com/facebookincubator/velox/pull/10979 fixed https://github.com/apache/incubator-gluten/issues/7194.
2. 9025 in branch-1.2 (https://github.com/oap-project/velox/commit/f34c9b1e264abb8e91563c5c358923b75276f5ec) is diff from https://github.com/facebookincubator/velox/pull/9025, which fixed coredump in new ut of 10979.

### CODE REVIEW

**Comment by ccat3z** [comment]
@weiting-chen @JkSelf Can you help to review this PR?

File: velox/exec/tests/WindowTest.cpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ TEST_F(WindowTest, rowBasedStreamingWindowOOM) {
 
       VELOX_ASSERT_THROW(
           readCursor(params, [](Task*) {}),
-          "Exceeded memory pool capacity after attempt to grow capacity through arbitration.");
+          "Exceeded memory pool cap of");
```
Comment (JkSelf) [inline velox/exec/tests/WindowTest.cpp:135]: @ccat3z Why need this change?

Comment (ccat3z) [reply velox/exec/tests/WindowTest.cpp:135]: https://github.com/facebookincubator/velox/pull/10506 changed exception message, which not included in branch 1.2

**Comment by weiting-chen** [comment]
The code looks fine to me.
The PR has also run pass 333 velox tests in jenkins with 1 failed test related to ParquetTableScanTest.timestampFilter test report empty parquet file, which is not related to the PR.
The issue is related to https://github.com/facebookincubator/velox/pull/4680 and not related to this PR.
And I will approval to merge it.

**Review by weiting-chen — APPROVED** [review]

**Comment by weiting-chen** [comment]
After merge the code, I will trigger another test in Gluten for branch-1.2 to see if there is any other issue.

**Pull request merged**